lib/managers/docker.js

```diff
@@ -754,11 +754,14 @@ export const extractTar = async (fullImageName, dir) => {
         preserveOwner: false,
         noMtime: true,
         noChmod: true,
-        strict: true,
+        strict: false,
         C: dir,
         portable: true,
-        onwarn: () => {
+        onwarn: (code, message) => {
           // ignore
+          if (code !== "TAR_ENTRY_INFO" || DEBUG_MODE) {
```

Comment:
I wouldn't say "a lot" but, yes, it will.  IMO, that is desirable for `DEBUG_MODE`.  The `tar` command in linux does the same thing.

Reply:
The problem with your approach is that it relies on the error message matching your expectations.  That creates a regression problem you're not prepared to address right now.  Your solution also didn't look at the `code` which is far more likely to remain unchanged with subsequent releases.

Reply:
Can we just make this if (DEBUG_MODE) ?

Reply:
You're suggesting that issues of types like `TAR_ENTRY_INVALID` should be simply ignored without any printed message unless the user enables DEBUG_MODE.  Skipping `TAR_ENTRY_INFO` is fine.  Skipping the other ones is NOT fine.  I can tell you that, if you did this, we would opt to not use `cdxgen` at all and go with `syft` or something else.  Simply black-holing warnings that are causing entries to not be expanded is a BIG DEAL.  That means that `cdxgen` is not scanning whatever wasn't expanded, and the user has no idea this happened.  This means that, technically, the current version of `cdxgen` already potentially has a security issue if people are relying on this SBOM for vulnerability assessment.  

Based on your other responses on this MR, I'm not sure you're realizing the potential scope of the problem or that you've yet conceded the current version of `cdxgen` is scarily broken with respect to scanning docker images.  The user currently has no idea that things were not expanded and there is no determinate pattern to what can be skipped.  The only reason we know about it now is because the expansion happened to not expand the `package.json` file.  But what if it failed to expand other files (OS, components, etc.) that the scanner then skips?  The user gets no errors, no warning messages and, most importantly, inaccurate SBOM output.

Reply:
Doing `strict: false` is actually a far bigger problem than making some warnings only show up in DEBUG_MODE. I have to balance the amount of messages cdxgen throws up in default mode, since our users would happily create an issue for each and every message, without the slightest hesitation.

It is work-in-progress, alright. There is only so much a small unfunded team could do. You can always feel free to go with another tool or rollup the sleeves and help us with dev and testing.